wallet-new/src/Cardano/Wallet/Kernel/Compat.hs

```diff
@@ -0,0 +1,103 @@
+-- | This module exports various tools for compatibility with old wallet
+-- dependencies.
+
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes                 #-}
+
+module Cardano.Wallet.Kernel.Compat
+  ( DBReadT(DBReadT, unDBReadT)
+  , withMonadDBRead
+  , getCoreConfigurations
+  ) where
+
+import           Control.Monad.IO.Unlift (MonadUnliftIO, UnliftIO (UnliftIO),
+                     askUnliftIO, unliftIO, withUnliftIO)
+import           Control.Monad.Trans.Class (MonadTrans)
+import           Control.Monad.Trans.Reader (ReaderT (ReaderT), runReaderT)
+import           Control.Monad.Trans.Resource (transResourceT)
+import           Data.Conduit (transPipe)
+import           Pos.Core (BlockVersionData, CoreConfiguration,
+                     GeneratedSecrets, GenesisData, GenesisHash (GenesisHash),
+                     ProtocolConstants, ProtocolMagic, Timestamp,
+                     generatedSecrets, genesisBlockVersionData, genesisData,
+                     genesisHash, getGenesisHash, protocolConstants,
+                     withCoreConfiguration, withCoreConfigurations,
+                     withGeneratedSecrets, withGenesisBlockVersionData,
+                     withGenesisData, withGenesisHash, withProtocolConstants)
+import           Pos.Core.Configuration (HasConfiguration)
+import           Pos.DB.Class (MonadDBRead (..), Serialized (Serialized))
+import           Universum
+
+import           Pos.DB.Block (getSerializedBlock, getSerializedUndo)
+import           Pos.DB.Rocks.Functions (dbGetDefault, dbIterSourceDefault)
+import           Pos.DB.Rocks.Types (MonadRealDB, NodeDBs)
+
+--------------------------------------------------------------------------------
+
+-- | This monad transformer exists solely to provide a 'MonadRealDB' instance,
+-- as required by upstream libraries.
+newtype DBReadT m a = DBReadT { unDBReadT :: ReaderT NodeDBs m a }
+  deriving (Functor, Applicative, Monad, MonadThrow, MonadTrans, MonadIO)
+
+instance MonadUnliftIO m => MonadUnliftIO (DBReadT m) where
+  askUnliftIO =
+    DBReadT (withUnliftIO (\u -> pure (UnliftIO (unliftIO u . unDBReadT))))
+
+instance (HasConfiguration, MonadThrow (DBReadT m), MonadRealDB NodeDBs (ReaderT NodeDBs m))
```

Comment:
This seems strange to me. You are requiring instances for the monad you are defining here...? Why not give a `MonadThrow` instance for `DBReadT`? (That would then probably reduce this constraint to `MonadThrow m`, which is more reasonable.)

Reply:
There is a `MonadThrow` instance already. This was just a mistake, I replaced this with `MonadThrow m`.